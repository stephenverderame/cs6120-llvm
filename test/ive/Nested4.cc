// NOLINTBEGIN
#include <cstdio>
int main()
{
    int m = 20;
    for (int i = 0; i < 10; ++i) {
        // ind var
        int p = m + i * 10;
        for (int j = 0; j < i; ++j) {
            // not ind var
            int k = j + i * p;
            for (int m = 0; m < k; m += 20) {
                int n = m + k * p;
                printf("%d\n", n);
            }

            for (int m = 0; m < k; m++) {
                // 1 ind var (m * p)
                int n = k - m * p;
                printf("%d\n", n);
            }
        }
    }
}
// NOLINTEND