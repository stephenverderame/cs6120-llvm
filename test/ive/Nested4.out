Loop: 
13:                                               ; preds = %20, %10
  %.01 = phi i32 [ 0, %10 ], [ %21, %20 ]
  %14 = icmp slt i32 %.01, %12
  br i1 %14, label %16, label %15

Basic IV:   %21 = add nsw i32 %.01, 20
Basic IV:   %.01 = phi i32 [ 0, %10 ], [ %21, %20 ]
Loop: 
23:                                               ; preds = %30, %22
  %.0 = phi i32 [ 0, %22 ], [ %31, %30 ]
  %24 = icmp slt i32 %.0, %12
  br i1 %24, label %26, label %25

Derived IV: %27  = .0 * %6 
Basic IV:   %31 = add nsw i32 %.0, 1
Basic IV:   %.0 = phi i32 [ 0, %22 ], [ %31, %30 ]
Loop: 
7:                                                ; preds = %37, %4
  %.03 = phi i32 [ 0, %4 ], [ %38, %37 ]
  %8 = icmp slt i32 %.03, %.02
  br i1 %8, label %10, label %9

Basic IV:   %38 = add nsw i32 %.03, 1
Basic IV:   %.03 = phi i32 [ 0, %4 ], [ %38, %37 ]
Loop: 
1:                                                ; preds = %40, %0
  %.02 = phi i32 [ 0, %0 ], [ %41, %40 ]
  %2 = icmp slt i32 %.02, 10
  br i1 %2, label %4, label %3

Basic IV:   %41 = add nsw i32 %.02, 1
Derived IV: %5  = .02 * i32 10
Derived IV: %6  = .02 * i32 10 + i32 20
Basic IV:   %.02 = phi i32 [ 0, %0 ], [ %41, %40 ]
